src/app/modules/form/public-form/__tests__/public-form.routes.spec.ts

```diff
@@ -0,0 +1,279 @@
+          spcpSession: { userName: MOCK_COOKIE_PAYLOAD.userName },
+          isIntranetUser: false,
+        }),
+      )
+
+      // Act
+      // Set cookie on request
+      const actualResponse = await request
+        .get(`/${formId}/publicform`)
+        .set('Cookie', ['jwtCp=mockJwt'])
+
+      // Assert
+      expect(actualResponse.status).toEqual(200)
+      expect(actualResponse.body).toEqual(expectedResponseBody)
+    })
+    it('should return 200 with public form when form has AuthType.MyInfo and valid formId', async () => {
+      // Arrange
+      const { form } = await dbHandler.insertEmailForm({
+        formOptions: {
+          esrvcId: 'mockEsrvcId',
+          authType: AuthType.MyInfo,
+          hasCaptcha: false,
+          status: Status.Public,
+        },
+      })
+      // NOTE: This is needed to inject admin info into the form
+      const fullForm = await dbHandler.getFullFormById(form._id)
+      const expectedResponseBody = JSON.parse(
+        JSON.stringify({
+          form: fullForm.getPublicView(),
+          spcpSession: { userName: 'S1234567A' },
+          isIntranetUser: false,
+        }),
+      )
+      const cookie = JSON.stringify({
+        accessToken: 'mockAccessToken',
+        usedCount: 0,
+        state: MyInfoCookieState.Success,
+      })
+
+      // Act
+      const actualResponse = await request
+        .get(`/${form._id}/publicform`)
+        .set('Cookie', [
+          // The j: indicates that the cookie is in JSON
+          `${MYINFO_COOKIE_NAME}=j:${encodeURIComponent(cookie)}`,
+        ])
+
+      // Assert
+      expect(actualResponse.status).toEqual(200)
+      expect(actualResponse.body).toEqual(expectedResponseBody)
+    })
+
+    it('should return 404 if the form does not exist', async () => {
+      // Arrange
+      const cookie = JSON.stringify({
+        accessToken: 'mockAccessToken',
+        usedCount: 0,
+        state: MyInfoCookieState.Success,
+      })
+      const MOCK_FORM_ID = _.pad('', 24, '1')
+      const expectedResponseBody = JSON.parse(
+        JSON.stringify({
+          message: 'Form not found',
+        }),
+      )
+
+      // Act
+      const actualResponse = await request
+        .get(`/${MOCK_FORM_ID}/publicform`)
+        .set('Cookie', [
+          // The j: indicates that the cookie is in JSON
+          `${MYINFO_COOKIE_NAME}=j:${encodeURIComponent(cookie)}`,
+        ])
+
+      // Assert
+      expect(actualResponse.status).toEqual(404)
+      expect(actualResponse.body).toEqual(expectedResponseBody)
+    })
+
+    it('should return 404 if the form is private', async () => {
+      // Arrange
+      const { form } = await dbHandler.insertEmailForm({
+        formOptions: { status: Status.Private },
+      })
+      const expectedResponseBody = JSON.parse(
+        JSON.stringify({
+          message:
+            'If you think this is a mistake, please contact the agency that gave you the form link.',
```

Comment:
The body response should have 2 more key-values:

```ts
formTitle: string
isPageFound: boolean
```

check if this is a regression

Reply:
good catch. this is a regression. will update tests and the controller